tools/ideabuck/src/com/facebook/buck/intellij/ideabuck/run/BuckBuildBeforeRun.java

```diff
@@ -0,0 +1,170 @@
+
+import javax.swing.Icon;
+import java.util.List;
+
+public class BuckBuildBeforeRun extends ExternalSystemBeforeRunTaskProvider {
+
+  private static final Key<ExternalSystemBeforeRunTask> ID = Key.create("Buck.BeforeRunTask");
+  private static final ProjectSystemId SYSTEM_ID = new ProjectSystemId("buck");
+
+  private String targetResult;
+  private Project currentProject;
+
+  public BuckBuildBeforeRun(@NotNull Project project) {
+    super(SYSTEM_ID, project, ID);
+  }
+
+  @Override
+  public String getName() {
+    return "Buck Build";
+  }
+
+  @Override
+  public String getDescription(ExternalSystemBeforeRunTask task) {
+    return "Buck Build";
+  }
+
+  @NotNull
+  @Override
+  public ExternalSystemBeforeRunTask createTask(@NotNull RunConfiguration runConfiguration) {
+    return new ExternalSystemBeforeRunTask(ID, SYSTEM_ID);
+  }
+
+  @Nullable
+  @Override
+  public Icon getIcon() {
+    return BuckIcons.BUCK_TOOL_WINDOW_ICON;
+  }
+
+  @Override
+  public boolean canExecuteTask(
+      RunConfiguration configuration, ExternalSystemBeforeRunTask beforeRunTask
+  ) {
+    if (!(configuration instanceof ApplicationConfiguration)) {
+      return false;
+    }
+    VirtualFile containingFile = ((ApplicationConfiguration) configuration).getMainClass().getContainingFile().getVirtualFile();
+    VirtualFile buckFile = BuckFileUtil.getBuckFile(containingFile);
+
+    return buckFile != null;
+  }
+
+  @Override
+  public boolean executeTask(
+      final DataContext context,
+      RunConfiguration configuration,
+      ExecutionEnvironment env,
+      ExternalSystemBeforeRunTask beforeRunTask
+  ) {
+    if (currentProject == null) {
+      currentProject = configuration.getProject();
+    }
+
+    if (configuration instanceof ApplicationConfiguration) {
+      VirtualFile containingFile = ((ApplicationConfiguration) configuration).getMainClass()
+          .getContainingFile().getVirtualFile();
+
+      VirtualFile buckFile = BuckFileUtil.getBuckFile(containingFile);
+
+      if (buckFile == null) {
+        return false;
+      }
+
+      BuckBuildManager buildManager = BuckBuildManager.getInstance(currentProject);
+      BuckQueryCommandHandler buckQueryCommandHandler =
+          new BuckQueryCommandHandler(
+              currentProject,
+              buckFile.getParent(),
+              BuckCommand.QUERY,
+              new Function<List<String>, Void>() {
+                @javax.annotation.Nullable
+                @Override
+                public Void apply(@javax.annotation.Nullable List<String> strings) {
+                  if (strings == null
+                      || strings.isEmpty()
+                      || strings.get(0) == null
+                      || strings.get(0).isEmpty()) {
+                    return null;
+                  }
+                  String targetResultCandidate = strings.get(0);
+                  for (String candidate : strings) {
+                    if (!candidate.contains("lint")) {
```

Comment:
Why?

Reply:
This was to filter out our own lint targets, and should take all targets instead.